polylogue/cli/completion_engine.py

```
@@ -0,0 +1,282 @@
+
+
+def _list_provider_names() -> List[str]:
+    return ["drive", *LOCAL_SYNC_PROVIDER_NAMES]
+
+
+class CompletionEngine:
+    def __init__(self, env: CommandEnv, parser: argparse.ArgumentParser) -> None:
+        self.env = env
+        self.parser = parser
+
+    def complete(self, shell: str, cword: int, words: Sequence[str]) -> List[Completion]:
```

Ruff (0.15.9)

[warning] Too many return statements (13 > 6)

(PLR0911)